whisper.py

```diff
@@ -53,110 +51,98 @@ def load_model(
-                    "temperature": segment.temperature,
-                    "avg_logprob": segment.avg_logprob,
-                    "compression_ratio": segment.compression_ratio,
-                    "no_speech_prob": segment.no_speech_prob,
-                    "words": words_to_add,
-                }
-            )
-        asset_id, _ = get_asset_info(input_path)
-        # Also added "carrierId" because the DAAN format requires it
-        transcript = {"carrierId": asset_id, "segments": segments_to_add}
-        end_time = (time.time() - start_time) * 1000
-
-        if w_device == "cuda":
-            max_mem_usage, gpu_limit = gpu_mem_measure.stop_measure_gpu_mem()
-            logger.info(
-                "Maximum GPU memory usage: %dMiB / %dMiB (%.2f%%)"
-                % (
-                    max_mem_usage,
-                    gpu_limit,
-                    (max_mem_usage / gpu_limit) * 100,
+    if not model:
+        logger.info("Model not passed as param, need to obtain it first")
+        model = load_model(MODEL_BASE_DIR, W_MODEL, W_DEVICE)
+    if W_DEVICE == "cpu":
+        logger.warning(f"Device selected is {W_DEVICE}: using a batch size of 1")
+
+    os.environ["PYTORCH_KERNEL_CACHE_PATH"] = MODEL_BASE_DIR
+    logger.info("Processing segments")
+
+    if W_DEVICE == "cuda":
+        gpu_mem_measure = GpuMemoryMeasure()
+        gpu_mem_measure.start_measure_gpu_mem()
+
+    segments, _ = model.transcribe(
+        input_path,
+        vad_filter=W_VAD,
+        beam_size=W_BEAM_SIZE,
+        best_of=W_BEST_OF,
+        batch_size=W_BATCH_SIZE if W_DEVICE == "cuda" else 1,
+        language="nl",  # TODO: experiment without language parameter specified (for programs with foreign speech)
+        word_timestamps=W_WORD_TIMESTAMPS,
+    )
+
+    segments_to_add = []
+    for segment in segments:
+        words_to_add = []
+        if W_WORD_TIMESTAMPS:
+            for word in segment.words:
+                words_to_add.append(
+                    {
+                        "text": word.word.strip(),
+                        "start": word.start,
+                        "end": word.end,
+                        "confidence": word.probability,
+                    }
                 )
+        segments_to_add.append(
+            {
+                "id": segment.id,
+                "seek": segment.seek,
+                "start": segment.start,
+                "end": segment.end,
+                "text": segment.text.strip(),
+                "tokens": segment.tokens,
+                "temperature": segment.temperature,
+                "avg_logprob": segment.avg_logprob,
+                "compression_ratio": segment.compression_ratio,
+                "no_speech_prob": segment.no_speech_prob,
+                "words": words_to_add,
+            }
+        )
+    # Also added "carrierId" because the DAAN format requires it
+    transcript = {"carrierId": asset_id, "segments": segments_to_add}
```

Comment:
This is essentially converting the raw segments to some transcript format, which seems an isolated responsibility. So I'd suggest to move this bit to a dedicated function.